llvm/lib/Target/DirectX/DXILIntrinsicExpansion.cpp

```diff
@@ -363,9 +382,13 @@ static bool expandIntrinsic(Function &F, CallInst *Orig) {
   case Intrinsic::dx_normalize:
     Result = expandNormalizeIntrinsic(Orig);
     break;
+  case Intrinsic::dx_fdot:
+    Result =
+        expandFloatDotIntrinsic(Orig, Orig->getOperand(0), Orig->getOperand(1));
```

Comment:
you are right that is a little weird maybe, a function overload would be better? Its a nitpick so feel free to ignore.

Reply:
Done